tests/config/basic/butler.yaml

```diff
@@ -3,8 +3,32 @@ datastore:
   cls: 'lsst.daf.butler.datastores.posixDatastore.PosixDatastore'
   root: './butler_test_repository'
   create: true
-  formatters:
-    SourceCatalog: "lsst.daf.butler.formatters.fitsCatalogFormatter.FitsCatalogFormatter"
+  storageClasses:
+    StructuredData:
+      # Data from a JSON or YAML file
+      type: "dict"
+      # formatter: "lsst.daf.butler.formatters.structuredDataFormatter.YAMLFormatter"
+    TablePersistable:
+      type: "lsst.afw.table"
+    Image:
+      type: "lsst.afw.image.MaskedImage"
+    Catalog:
+      type: "lsst.afw.table.BaseCatalog"
+    SourceCatalog:
+      type: "lsst.afw.table.SourceCatalog"
+      formatter: "lsst.daf.butler.formatters.fitsCatalogFormatter.FitsCatalogFormatter"
+    Exposure:
+      type: "lsst.afw.image.Exposure"
+      components:
+        image: "Image"
+        mask: "Image"
+        variance: "Image"
+        wcs: "TablePersistable"
+        psf: "TablePersistable"
+        photoCalib: "TablePersistable"
+        visitInfo: "TablePersistable"
+        apCorr: "TablePersistable"
+        coaddInputs: "TablePersistable"
```

Comment:
I can remove the quotes. Making them YAML references is something I'd have to look at (currently I need them to be strings to look up in the registry rather than items to look up from the YAML tree).